torchtitan/components/checkpoint.py

```diff
@@ -35,6 +35,14 @@
 from torchtitan.tools.logging import logger
 from torchtitan.tools.utils import GarbageCollection
 
+try:
+    # allow unpicking of WeightWithDynamicFloat8CastTensor, which are saved
+    # when doing FP8 training
+    from torchao.float8.fsdp_utils import WeightWithDynamicFloat8CastTensor
+    torch.serialization.add_safe_globals([WeightWithDynamicFloat8CastTensor])
+except:
+    pass
```

Comment:
Can we log a warning if this fails to reduce confusion for users if some future change to torch.serialization etc prevents this from being successful?